mlir/tools/miopen-gen/CMakeLists.txt

```diff
@@ -5,9 +5,16 @@ set(LLVM_LINK_COMPONENTS
 get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
 get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
 
+if(MLIR_INCLUDE_TESTS)
```

Comment:
This set of changes, with the one for mlir-miopen-driver, fixes compiling with -DMLIR_INCLUDE_TESTS=OFF.  It's the same pattern used in external/llvm-project/mlir/tools/*/CMakeLists.txt.

The default is ON, but I was trying to make sure my tests didn't cause trouble in the other case.  As it happens, the whole test directory is omitted when OFF, so my concern was a non-problem, but I found this build failure.